pkg/proxy/iptables/proxier.go

```diff
@@ -395,6 +395,7 @@ type iptablesJumpChain struct {
 var iptablesJumpChains = []iptablesJumpChain{
 	{utiliptables.TableFilter, kubeExternalServicesChain, utiliptables.ChainInput, "kubernetes externally-visible service portals", []string{"-m", "conntrack", "--ctstate", "NEW"}},
 	{utiliptables.TableFilter, kubeExternalServicesChain, utiliptables.ChainForward, "kubernetes externally-visible service portals", []string{"-m", "conntrack", "--ctstate", "NEW"}},
+	{utiliptables.TableFilter, kubeNodePortsChain, utiliptables.ChainInput, "kubernetes health check service portals", nil},
```

Comment:
You shouldn't need to add this; inbound connections already eventually pass through `kubeNodePortsChain` via the "tail-call" created at the end of `syncProxyRules`.

Reply:
Is that still true when the default `INPUT` chain policy is `DROP`?

Reply:
I thought it would work without this rule in my early attempts but turned out it didn't. The fact is for `NodePort` services, the traffic was `DNAT`ted in `nat` table, basically in this order:
```
-A PREROUTING -m comment --comment "kubernetes service portals" -j KUBE-SERVICES
-A KUBE-SERVICES -d 10.102.149.66/32 -p tcp -m comment --comment "default/nginx cluster IP" -m tcp --dport 8080 -j KUBE-SVC-2CMXP7HKUVJN7L6M
-A KUBE-SVC-2CMXP7HKUVJN7L6M -m comment --comment "default/nginx" -j KUBE-SEP-V6JOQ423IM4BM4HR
-A KUBE-SEP-V6JOQ423IM4BM4HR -s 100.96.0.11/32 -m comment --comment "default/nginx" -j KUBE-MARK-MASQ
-A KUBE-SEP-V6JOQ423IM4BM4HR -p tcp -m comment --comment "default/nginx" -m tcp -j DNAT --to-destination 100.96.0.11:80
```
In the rules above, `100.96.0.11` is the `nginx` pod IP.
In `filter` table, the traffic is accepted by the `KUBE-FORWARD` chain, see the third rule:
```
➜  ~ iptables -L KUBE-FORWARD --line-n -v
Chain KUBE-FORWARD (1 references)
num   pkts bytes target     prot opt in     out     source               destination
1        0     0 DROP       all  --  any    any     anywhere             anywhere             ctstate INVALID
2        0     0 ACCEPT     all  --  any    any     anywhere             anywhere             /* kubernetes forwarding rules */ mark match 0x4000/0x4000
3      235 42692 ACCEPT     all  --  any    any     anywhere             anywhere             /* kubernetes forwarding conntrack pod source rule */ ctstate RELATED,ESTABLISHED
4        0     0 ACCEPT     all  --  any    any     anywhere             anywhere             /* kubernetes forwarding conntrack pod destination rule */ ctstate RELATED,ESTABLISHED
```
So, different from healthcheck nodeports, the traffic towards node ports got `DNAT`ted and instead of hitting `INPUT` chain, it will hit `FORWARD` chain. However health check node ports are listening on local addresses so the incoming packets will hit `INPUT` chain.

Before this change, there's no `KUBE-NODEPORTS` chain in the `filter` table at all, so whether the traffic is accepted or not is purely depending on the policy of `INPUT` chain. That's why we need to add a rule from `INPUT` chain to the newly added `KUBE-NODEPORTS` chain.

Speaking of the difference of node port traffic that is `DNAT`ted and the traffic towards health check endpoints that listens on local ports, I realized that if we have pods on `hostNetwork`, then node ports towards them won't work just like for health check node ports. I just validated that in my environment and the issue is really there, and I'm confident that I could fix that using the same approach just like for health check node port. I don't want to make more changes in the PR so maybe open an issue and fix it in another one, what do you guys think? @andrewsykim @danwinship ?

Reply:
I think that better to do that in a follow up PR,

Reply:
Ah... there are two completely separate `KUBE-NODEPORTS` chains now... one in `-t nat` and one in `-t filter`

Reply:
ok, I think this is right, but don't use the word "portals" in the description. That's archaic pre-kube-1.0 terminology. Actually, just changing it to "ports" is probably what you meant anyway.

Reply:
Sure, changed here.